net/redisclient.go

```diff
@@ -30,7 +30,7 @@ type RedisOptions struct {
 	// AddrUpdater is a func that is regularly called to update
 	// redis address list. This func should return a list of redis
 	// shards.
-	AddrUpdater func() []string
+	AddrUpdater []string
```

Comment:
This will leave us with a static list, which we don't need. We need it to be a function to call it every time, we just want the function return type to be `[]string, error` so 

```suggestion
	AddrUpdater func() ([]string, error)
```
Same with others